### DISCUSSION TITLE
Export to PPROF format

### DISCUSSION THREAD

**Discussion by adrien-f** [NoiseByNorthwest/php-spx#179] [Ideas]
Greetings :wave: !

I was wondering if either anyone tried to convert the data generated by SPX to the [PPROF](https://github.com/google/pprof) format or if it is something on the roadmap for the project? 

Some resources:

* https://github.com/google/pprof
* NodeJS implementation: https://github.com/google/pprof-nodejs

**Comment by NoiseByNorthwest** [comment]
Hi,

In early versions it supported 2 external formats, Callgrind & Google's Trace Event Format but with the arrival of the web UI I've dropped these supports with no plan to reintroduce them.

The main reason is that since the web UI is better than K/QCachegrind or Chromium's about:tracing then I should spent time on more valuable things.

However I could easily restore these 2 format supports on web UI side, as "export as ..." buttons, because there're both easy to implement in vanilla modern JS.

But regarding the PPROF format it does not seem to be easy to implement, espacially because it requires a protobuf infrastructure.

**Comment by NoiseByNorthwest** [comment]
Web UI now supports exporting full SPX reports to pprof and callgrind formats ([v0.5](https://github.com/NoiseByNorthwest/php-spx/tree/v0.5) branch).
